SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc

```diff
@@ -56,6 +57,19 @@ G4VPhysicalVolume *DDG4Builder::BuildGeometry(SensitiveDetectorCatalog &catalog)
     auto fff = it.first->GetName();
     catalog.insert({sClassName.data(), sClassName.size()}, {sROUName.data(), sROUName.size()}, fff);
 
+    auto reflectedG4LogicalVolumeName = fff + "_refl";
+    bool hasReflectedVolumeInStore = false;
+    G4LogicalVolumeStore *theStore = G4LogicalVolumeStore::GetInstance();
+    for (const auto &lv : *theStore) {
+      if (lv->GetName().find(reflectedG4LogicalVolumeName) != std::string::npos) {
+        hasReflectedVolumeInStore = true;
```

Comment:
Shouldn't you add after this line:
```
break;
```

Reply:
yes need to clean style a bit, will slightly improve.

Reply:
But for now I was focused on fixing the issue, there was no point improving performance on wrong solutions.

Reply:
break is a good point, I will add it

Reply:
@cvuosalo Yep I remember what happened now, actually I wanted to use a std::find, but I did not know what was the type of G4LogicalVolumeStore, so just made the easy loop, to first see if the fix was working anyway.
Just looked it up, G4LogicalVolumeStore is a std::vector<G4LogicalVolume*>.
I have just used a std::find_if in https://github.com/cms-sw/cmssw/pull/32241/ , it is equivalent to a loop and break, as of course stops at first occurrence.